src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexProcessingJob.cs

```diff
@@ -323,14 +324,53 @@ public async Task ProcessSearchResultsAsync(SearchResult results, IReadOnlyDicti
         }
 
         /// <summary>
-        /// Determines whether a continuation job should be created based on the search results.
+        /// Determines whether a continuation job should be created.
+        /// If the current max resource surrogate ID is less than the end surrogate ID of the job definition, then a continuation job is needed to finish reindexing.
+        /// If there are other jobs in the same group with a start surrogate ID equal to the next surrogate ID to process, then a continuation job is not needed (as the continuation job was already created).
         /// </summary>
         /// <param name="result">The search result from the current processing batch</param>
         /// <returns>True if a continuation job is needed, false otherwise</returns>
-        private bool ShouldCreateContinuationJob(SearchResult result)
+        private async Task<bool> ShouldCreateContinuationJobAsync(SearchResult result)
         {
-            return result?.MaxResourceSurrogateId > 0 &&
-                   result.MaxResourceSurrogateId < _reindexProcessingJobDefinition.ResourceCount.EndResourceSurrogateId;
+            bool isContinuationJobRequired = result.MaxResourceSurrogateId > 0 || result.MaxResourceSurrogateId < _reindexProcessingJobDefinition.ResourceCount.EndResourceSurrogateId;
+
+            // Check if there are more resources to process in the current range.
+            if (!isContinuationJobRequired)
+            {
+                return false;
+            }
+
+            // Get all jobs in this group.
+            IReadOnlyList<JobInfo> jobs = await _queueClient.GetJobByGroupIdAsync((byte)QueueType.Reindex, _jobInfo.GroupId, false, CancellationToken.None);
+            if (jobs == null || jobs.Count == 0)
```

Comment:
This shouldn't be possible. At minimum the job we are processing is in the group.

Reply:
I was thinking about leaving this just a security case. But that's fine.